app/models/health_care_application.rb

```diff
@@ -248,27 +248,36 @@ def submit_async
     self
   end
 
-  def log_submission_failure
+  def log_sync_submission_failure
+    StatsD.increment("#{HCA::Service::STATSD_KEY_PREFIX}.sync_submission_failed")
+    StatsD.increment("#{HCA::Service::STATSD_KEY_PREFIX}.sync_submission_failed_short_form") if short_form?
+    log_submission_failure_details
+  end
+
+  def log_async_submission_failure
     StatsD.increment("#{HCA::Service::STATSD_KEY_PREFIX}.failed_wont_retry")
     StatsD.increment("#{HCA::Service::STATSD_KEY_PREFIX}.failed_wont_retry_short_form") if short_form?
+    log_submission_failure_details
+  end
 
-    if form.present?
-      PersonalInformationLog.create!(
-        data: parsed_form,
-        error_class: 'HealthCareApplication FailedWontRetry'
-      )
+  def log_submission_failure_details
+    return if form.blank?
```

Comment:
I thought a guard clause here was clearer/more succinct than wrapping it in the if conditional.

Reply:
Yeah that is weird. Let me look into changing it to `parsed_form` to make it clearer, since that it what this really cares about. It _should_ be the same since parsed_form is `@parsed_form ||= JSON.parse(form)`, although we do the overrides logic on the parsed_form so not sure if that does something to it.

Reply:
Yeah, the logic around `parsed_form` is a little confusing to me at the moment. TBH, I'm a little hesitant to introduce too many changes here because I am not confident there is proper test coverage, but I do think this could do with some refactoring to make it clearer/ensure we are doing what we think we are doing. It does look lik, the form should _always_ exist here because of this validation, right? `  validates(:form, presence: true, on: :create)`? I think I'd probably prefer to leave it as is atm, but I definitely would like to refactor some of this to better test it and make it easier to read.